dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/ssh/SSHSessionHolder.java

```diff
@@ -0,0 +1,348 @@
+
+            if (maxFileSize >= 0) {
+                if (totalSize > (long) maxFileSize * KB_UNIT * KB_UNIT) {
+                    customLogger.error("The size of :{} has exceeded the maximum size:{}, size: {}", totalSize,
+                            maxFileSize, localDirPath);
+                    return false;
+                }
+            }
+
+            if (file.isDirectory()) {
+                File[] files = file.listFiles();
+                if (files == null || files.length == 0) {
+                    customLogger.error("{} is a empty directory.", localDirPath);
+                    return false;
+                }
+                for (File subFile : files) {
+                    String sf = subFile.getCanonicalPath();
+                    if (subFile.isDirectory()) {
+                        String mkdirPath = remoteDirPath + "/" + subFile.getName();
+                        try {
+                            channelSftp.cd(mkdirPath);
+                        } catch (SftpException e) {
+                            if (!createDirOnRemote(mkdirPath)) {
+                                customLogger.error("Could not create directory {} on remote session:{}", mkdirPath,
+                                        sessionHost.toString());
+                                return false;
+                            }
+                        }
+                        if (!sftpDir(createChannelSftp(), sf, mkdirPath, enableUploadMonitor, maxUploadRate, -1,
+                                customLogger)) {
+                            customLogger.error("sftp {} to {}:{} failed.", sf, sessionHost.toString(), mkdirPath);
+                            return false;
+                        }
+                    } else {
+                        upload(channelSftp, sf, remoteDirPath, enableUploadMonitor, maxUploadRate);
+                    }
+                }
+            } else {
+                upload(channelSftp, file.getCanonicalPath(), remoteDirPath, enableUploadMonitor, maxUploadRate);
+            }
+            return true;
+        } catch (Exception e) {
+            throw new SSHException(
+                    "sftp " + localDirPath + " to " + sessionHost.toString() + ":" + remoteDirPath + " failed.", e);
+        }
+    }
+
+    /**
+     * Create a directory on the remote server
+     * Because JSch does not support one-time creation of multi-layer directories, so just use `mkdir -p` instead
+     * @param remoteDirPath remote directory path
+     * @return result
+     */
+    public boolean createDirOnRemote(String remoteDirPath) {
+        logger.info("create directory:{} on remote:{}", remoteDirPath, sessionHost.toString());
+        SSHResponse response = execCommand("mkdir -p " + remoteDirPath);
+        return response.getExitCode() == 0;
+    }
+
+    public void clearPath(String path) {
+        if (SINGLE_SLASH.equals(path)) {
+            return;
+        }
+        execCommand("rm -rf " + path);
```

Comment:
I think it's better to use `File.delete` to perform operations instead of shell commands.

Reply:
> I think it's better to use `File.delete` to perform operations instead of shell commands.

Becuase these codes need to clear the path on the remote SSH server, so use the `rm -rf` command, not `File.delete`

Reply:
Please don't mark it as resolved if it's not addressed. cc @zhongjiajie @caishunfeng

Reply:
> Please don't mark it as resolved if it's not addressed. cc @zhongjiajie @caishunfeng

Sorry, sure, and i think that `File.delete` cannot delete the remote dir and files, so i think that this case has resolved. WDYT.